File: Examples/BroadcastReceiverCompoundOrderedBroadcastWithResultReceiver/src/course/examples/BroadcastReceiver/CompoundOrderedBroadcast/Receiver3.java
package course.examples.BroadcastReceiver.CompoundOrderedBroadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;


public class Receiver3 extends BroadcastReceiver {

	private final String TAG = "Receiver3";

	@Override
	public void onReceive(Context context, Intent intent) {

		Log.i(TAG, "INTENT RECEIVED by Receiver3");

		String tmp = getResultData() == null ? "" : getResultData();
		setResultData(tmp + ":Receiver 3");
		
	}

}

File: ExamplesKotlin/BcastRecCompOrdBcastWithResRec/app/src/main/java/course/examples/broadcastreceiver/resultreceiver/Receiver3.kt
package course.examples.broadcastreceiver.resultreceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log


class Receiver3 : BroadcastReceiver() {

    companion object {
        private const val TAG = "Receiver3"
    }

    override fun onReceive(context: Context, intent: Intent) {

        Log.i(TAG, "INTENT RECEIVED by Receiver3")

        val tmp = if (resultData == null) "" else resultData
        resultData = "$tmp:Receiver 3"

    }

}
